test/e2e/mongodbtests/mongodbtests.go

```diff
@@ -74,21 +74,41 @@ func statefulSetIsNotReady(mdb *mdbv1.MongoDBCommunity, interval time.Duration,
 
 func StatefulSetHasOwnerReference(mdb *mdbv1.MongoDBCommunity, expectedOwnerReference metav1.OwnerReference) func(t *testing.T) {
```

Comment:
IMO this will be a redundant since we'll actually not test our functionality but the one k8s has. If we check that these resources were created and they have the correct OwnerReference, then the expectation would be that k8s would delete them if their owner reference is deleted.
Or maybe there are additional checkes we can get from this proposed e2e test?

Reply:
Generally, we write the E2E test just to automate the manual test we would be doing anyway to test the workflow.

But in this particular case, I think it's fine to skip that.